### PR TITLE
feat: add data api role arn in the output of redshift stack 

### PR DESCRIPTION
**Pull request by llmin**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

update redshift stack - add data api role to stack output
update quicksight stack 
- update the output of QuickSight dashboard

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy control plane with CloudFront + S3 + API gateway
  - [ ] deploy control plane within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

### CODE REVIEW

File: src/analytics/analytics-on-redshift.ts [hunk 1]
```diff
@@ -74,11 +75,14 @@ export class RedshiftAnalyticsStack extends NestedStack {
       throw new Error('Must specify ONLY one of new Redshift Serverless, existing Redshift Serverless or Provioned Redshift.');
     }
 
+    if (!props.newRedshiftServerlessProps && !props.existingRedshiftServerlessProps && !props.provisionedRedshiftProps ) {
```
Comment (zxkane) [inline src/analytics/analytics-on-redshift.ts:78]: This case should already be covered by above check.

File: src/common/constant.ts [hunk 2]
```diff
@@ -278,6 +278,8 @@ export const OUTPUT_DATA_MODELING_REDSHIFT_SERVERLESS_NAMESPACE_NAME =
   'StackCreatedRedshiftServerlessNamespaceName';
 export const OUTPUT_DATA_MODELING_REDSHIFT_BI_USER_CREDENTIAL_PARAMETER_SUFFIX =
   'BIUserCredentialParameterName';
+export const REDSHIFT_DATA_API_ROLE_ARN =
```
Comment (zxkane) [inline src/common/constant.ts:281]: follw the naming pattern like `OUTPUT_DATA_MODELING_REDSHIFT_BI_USER_CREDENTIAL_PARAMETER_SUFFIX`

File: src/data-reporting-quicksight-stack.ts [hunk 3]
```diff
@@ -187,6 +194,11 @@ export class DataReportingQuickSightStack extends Stack {
       value: dababoards,
     }).overrideLogicalId('Dashboards');
 
+    new CfnOutput(this, 'InternalUser', {
```
Comment (zxkane) [inline src/data-reporting-quicksight-stack.ts:197]: manage the output name in constant to be shared with backend api

File: src/data-reporting-quicksight-stack.ts [hunk 4]
```diff
@@ -174,6 +174,13 @@ export class DataReportingQuickSightStack extends Stack {
     cr.node.addDependency(vPCConnectionResource);
     cr.node.addDependency(template);
 
+    const userCustomResource = createInternelUserCustomResource(this, {
```
Comment (zxkane) [inline src/data-reporting-quicksight-stack.ts:177]: Is it duplicated with the work in #114 ? @tyyzqmf

File: src/reporting/lambda/custom-resource/user/index.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+  } catch (err) {
+
+    if (err instanceof ResourceNotFoundException) {
+      logger.warn(`user ${email} not exist, will create it`);
+    } else {
+      throw err;
+    }
+  }
+
+  const command = new RegisterUserCommand({
```
Comment (zxkane) [inline src/reporting/lambda/custom-resource/user/index.ts:85]: Does it mean each pipeline will create a such user for manaing the dashboards/visuals?

Comment (zxkane) [reply src/reporting/lambda/custom-resource/user/index.ts:85]: I prefer to manage this user in backend for all pipelines.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/data-reporting-quicksight-stack.ts (hunk 4 above)
Comment (llmin) [reply src/data-reporting-quicksight-stack.ts]: not duplicate with work in #114

File: src/reporting/lambda/custom-resource/user/index.ts (hunk 5 above)
Comment (llmin) [reply src/reporting/lambda/custom-resource/user/index.ts:85]: after talk with @tyyzqmf , will move internal user creation to backend, and create this internal user before call stacks if QuickSight reporting function is enabled.

File: src/data-reporting-quicksight-stack.ts (hunk 3 above)
Comment (llmin) [reply src/data-reporting-quicksight-stack.ts:196]: this output is not necessary after move internal use creation out of quicksight stack.

File: src/data-reporting-quicksight-stack.ts (hunk 4 above)
Comment (tyyzqmf) [reply src/data-reporting-quicksight-stack.ts]: This user is preferably created by backend when the pipeline is creating, if QuickSight is enabled for the pipeline.

File: src/reporting/lambda/custom-resource/quicksight/index.ts [hunk 6]
```diff
@@ -372,8 +372,8 @@ const createDataSet = async (quickSight: QuickSight, awsAccountId: string, princ
     let logicalMap = undefined;
     if (needLogicalMap) {
       logicalMap = {
-        LogialTable1: {
-          Alias: 'Alias_LogialTable1',
+        LogicalTable1: {
```
Comment (zxkane) [inline src/reporting/lambda/custom-resource/quicksight/index.ts:375]: Is it a field of QuickSight's component? How did it work when we used a wrong field name?

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/reporting/lambda/custom-resource/quicksight/index.ts (hunk 6 above)
Comment (llmin) [reply src/reporting/lambda/custom-resource/quicksight/index.ts:375]: can be any string, this string will be displayed when you edit dataset
截屏2023-08-09 19 27 19

**Review by zxkane — APPROVED** [review]

**Pull request merged**